web/packages/teleport/src/services/bot/consts.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiBot, FlatBot } from 'teleport/services/bot/types';
+
+export function makeListBot(bot: ApiBot): FlatBot {
```

Comment:
for any arrays (and objects while i'm at it), i assigned the expected empty values at this level, because backend doesn't guarantee that an `empty array` returns as `[]`, it can be `null`. i wasn't sure if this will be an issue for this api, but just FYI